src/components/constants/Pokemon.ts

```diff
@@ -128,6 +128,51 @@ export interface IPokemonMoveListing {
 }
 
 /**
+ * Data regarding requirements for a Pokemon's evolution
+ */
+export interface IPokemonEvolution {
+    /**
+     * The name of the Pokemon that this Pokemon evolves into.
+     */
+    evolvedForm: string;
+
+    /**
+     * A collection of the requirements for the Pokemon to evolve.
+     */
+    requirements: IPokemonEvolutionRequirements[];
+}
+
+/**
+ * The requirements for a Pokemon to be able to evolve.
+ */
+export interface IPokemonEvolutionRequirements {
```

Comment:
Ahh, this isn't how discriminated union types work. You'll want to declare one interface for each type, then a type equivalent to the `|` (union) of all of them.